### ISSUE TITLE
datalayer type circle expands cirles to full size  at certain zoom level

### ISSUE THREAD

**Issue by laudaget** [mapbox/mapbox-gl-js#12854]
After upgrading my mapbox v2.15 to mapbox v3 to get the new 3d map the circles from my datalayer filled up the whole map. I cant see anything but the colors of the circles at a certain zoom level.

**mapbox-gl-js version**: [v3.0.0-beta.1](https://github.com/mapbox/mapbox-gl-js/releases/tag/v3.0.0-beta.1)

**browser**: Chrome v115.0.5790.171, Firefox v116.0.2

### Steps to Trigger Behavior

 1. Create a 3D map 
 2. Add a source and a layer of type circles to the map 
 3. Zoom into the circles at the map

[JSFiddle](https://jsfiddle.net/yzf3xbdh/5/)

### Expected Behavior
- Circles should remain at every zoom level
### Actual Behavior
- Circles disappear at a zoom level 6 or higher

**Comment by jaspervanmegroot** [comment]
Same issue here, its not the zoom level but the Z index.

[](https://streamable.com/uvefmo)

**Comment by laudaget** [comment]
> Same issue here, its not the zoom level but the Z index. See the video for the example
> 
> https://streamable.com/uvefmo

@jaspervanmegroot yes you are right, its not about the zoom level. Its about the view angle. Thank you.

**Comment by andrewharvey** [comment]
Yeah I can confirm I noticed the same bug, when varying the map pitch up to 0 (straight down).

**Comment by stepankuzmin** [comment]
Hey all! This is a known issue, and it'll be fixed in the upcoming beta.

**Comment by jaspervanmegroot** [comment]
Hi @stepankuzmin, could you please provide an update on when the next beta will be available?
